src/server/api/mutations/updateContactTags.js

```diff
@@ -25,29 +26,19 @@ export const updateContactTags = async (
       campaign.organization_id
     );
 
-    // The rest is for ACTION_HANDLERS
-    const promises = [];
-    const getHandlersPromise = ActionHandlers.getActionHandlersAvailableForTagUpdate(
+    const handlers = await ActionHandlers.getActionHandlersAvailableForTagUpdate(
       organization,
       user
-    ).then(supportedActionHandlers => {
-      supportedActionHandlers.forEach(handler => {
-        const tagUpdatePromise = handler
-          .onTagUpdate(tags, user, contact, campaign, organization)
-          .catch(err => {
-            // eslint-disable-next-line no-console
-            console.error(
-              `Error executing handler.onTagUpdate for ${handler.name}, campaignContactId ${campaignContactId} error ${err}`
-            );
-          });
-        promises.push(tagUpdatePromise);
+    );
+    if (handlers && handlers.length > 0) {
+      // TODO: see if there is a better way to avoid unnecessary dispatches
```

Comment:
this looks good, though I wonder if we could pass the handlers list by-key maybe -- i.e. currently getActionHandlersAvailableForTagUpdate gets the loaded objects, but maybe it should just return the keys, which you can pass to the dispatchTask.